Makefile

```diff
@@ -40,15 +40,20 @@ DONT_FIND := -name tools -prune -o -name vendor -prune -o -name .git -prune -o -
 APP_GO_FILES := $(shell find . $(DONT_FIND) -name .y.go -prune -o -name .pb.go -prune -o -name cmd -prune -o -type f -name '*.go' -print)
 
 # Build flags
+ifeq ($(GO111MODULE), on)
+MOD_FLAG := -mod vendor
+VPREFIX := github.com/prometheus/common/version
+else
 VPREFIX := github.com/grafana/loki/vendor/github.com/prometheus/common/version
+endif
```

Comment:
Instead of detecting vendoring (which is tricky and will change again in Go 1.14), I suggest to set build values to Loki-specific variables (say in pkg/loki/build.go), and then pass them to Prometheus version package in `init` method.

Reply:
@slim-bean  Oh i get it, i would update until your pr merged.